libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1498,6 +1498,30 @@ void TypeChecker::endVisit(BinaryOperation const& _operation)
 {
 	TypePointer const& leftType = type(_operation.leftExpression());
 	TypePointer const& rightType = type(_operation.rightExpression());
+
+	if (_operation.getOperator() == Token::SHL || _operation.getOperator() == Token::SAR)
```

Comment:
Can you move this into `IntegerType::binaryOperatorResult`?

Reply:
Sure